### DISCUSSION TITLE
Set up AppImage builds

### DISCUSSION THREAD

**Discussion by gornostal** [Ulauncher/Ulauncher#888] [Packaging]
> [AppImage](http://appimage.org/) is an upstream packaging format and a set of tools with the main idea -- one app = one file. It bundles all the dependencies in a single executable file that can be installed on any Linux distro.

Having looked at AppImage I think it can be the ultimate solution to the problem of packaging.
However it's not that trivial to package Ulauncher with all its GTK and Python deps.
It just takes time to fix all the issues along the way.

Any help is greatly appreciated.

If somebody had experience with AppImage, please ping me in gitter: https://gitter.im/Ulauncher/General

**Comment by gornostal** [comment]
Spent another two days playing with AppImage.
Got Ulauncher to run actually. However run into an error with libwebkit2gtk:

```
** (utest:31453): WARNING **: Failed to load shared library 'libwebkit2gtk-4.0.so.37' referenced by the typelib: libwebp.so.5: cannot open shared object file: No such file or directory
/tmp/.mount_RIS3iC/usr/lib/python2.7/site-packages/ulauncher/ui/windows/PreferencesUlauncherDialog.py:81: Warning: cannot retrieve class for invalid (unclassed) type 'void'
  self.webview = WebKit2.WebView()
```

Also noticed startup time is much slower comparing to installations from deb/rpm.

I'm personally putting this on hold.
If anybody wants to take over AppImage integration, I stashed my code to `AppImage` branch.

**Comment by Justinzobel** [comment]
So I see a script to create an AppImage but I don't see one available for download? Was this discontinued?

**Comment by gornostal** [comment]
@Justinzobel I don't know where you found a script for AppImage built :)
AppImage support has never been finished. I've decided to go with DEB, RPM, and Arch Linux packages for now

**Comment by apfelchips** [comment]
I'm very interested in an AppImage of ULauncher. Could you please have another look?
